product_pricelist_direct_print_xlsx/wizards/product_pricelist_print.py

```diff
@@ -13,6 +14,32 @@ class ProductPricelistPrint(models.TransientModel):
 
     show_internal_category = fields.Boolean(string="Show internal categories")
 
+    print_child_categories = fields.Boolean(default=True, string="Print child categories")
+
+    def get_products_domain(self):
+        domain = super().get_products_domain()
+
+        if self.print_child_categories and self.categ_ids:
+            new_domain = []
+
+            for condition in domain:
+                if (
+                    isinstance(condition, tuple) and
+                    len(condition) == 3 and
+                    condition[0] == 'categ_id' and
+                    condition[1] == 'in'
+                ):
+                    aux_domain = expression.OR([
+                        [("categ_id", "in", self.categ_ids.ids)],
+                        [("categ_id", "child_of", self.categ_ids.ids)],
```

Comment:
Any reason why using both of them. If I'm correct, you want to search all children categories (including itself). If so, I would use child_of